Sephora-M/assignment4/Map.java

```diff
@@ -0,0 +1,109 @@
+package assignment4;
+
+import java.awt.Point;
+import java.util.ArrayList;
+
+
+/**Grid of booleans representing a map. True elements in the grid represent land whereas false tiles represent water.  
+ * 
+ * @author Sephora-M
+ */
+public class Map {
+	private boolean[][] grid;
+	private final int numRows;
+	private final int numCols;
+	
+	/**
+	 * This constructor takes a map of boolean representing the final map.
+	 * @param map, 2d array of boolean, represent the final map
+	 */
+	public Map(boolean[][] map) {
+		if (map == null || map[0].length == 0)
+			throw new IllegalArgumentException("Cannot instatiate an empty or a null grid.");
+		this.grid = map;
+		numCols = map.length;
+		numRows = map[0].length;
+	}
+	
+	/**
+	 * This constructor takes the size of the map and the locations of land tiles.
+	 * @param numRows int, the width of the map
+	 * @param numCols int, the height of the map
+	 */
+	public Map(int numRows, int numCols, ArrayList<Point> landTiles){
+		boolean[][] map = new boolean[numCols][numRows];
+		this.grid = map;
+		this.numCols = numCols;
+		this.numRows = numRows;
+		setLand(landTiles);
+	}
+
+	/**
+	 * Count the number of island in the map.
+	 * @return a int, number of island found
+	 */
+	public int countIslands(){
+		int numIslands = 0;
+		boolean[][] visited = new boolean[numCols][numRows];
```

Comment:
Another optimisation could be instead of marking Points as visited in the second array, you could remove land tiles from the original grid. :)

Reply:
Ah yes, good idea. I'll try and find time to do it this week.

Reply:
Didn't do this change in the end.. Hopefully I'll get away with it by saying that I prefer a solution that does not change the inputs 😉